### PR TITLE
SISRP-29888 Diagnostic & admin tools for Torquebox internals

### PR DESCRIPTION
**Pull request by raydavis**
https://jira.berkeley.edu/browse/SISRP-29888

Also includes support for RubyMine debugging of a Torquebox deployment, although it shouldn't be attempted without a lot of free memory & some tolerance for post-run ``jps -mlv`` & ``kill -9`` chores.

### CODE REVIEW

File: lib/background_job.rb [hunk 1]
```diff
@@ -31,15 +31,16 @@ module BackgroundJob
...
   #
   include TorqueBox::Messaging::Backgroundable
+  include ClassLogger
 
   attr_reader :background_job_id
 
   def self.unique_job_id
     15.times do
-      cache_key_candidate = "#{Time.now.to_f.to_s.gsub('.', '')}-#{SecureRandom.hex(8)}"
+      cache_key_candidate = "#{DateTime.now.to_s}-#{SecureRandom.hex(8)}"
```
Comment (raydavis) [inline lib/background_job.rb]: The change is meant to make the IDs more meaningful to Ops and DevOps staff when diagnosing a general slowdown. "Have the jobs been waiting for thirty seconds or for two hours?"

File: lib/torquebox_tester.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  def start_background_and_wait(times, pauses)
+    logger.warn "#{@background_job_id} About to start twiddling and will not return until done"
+    background_job_set_total_steps times
+    tracker = background_correlate(background.twiddle_thumbs(@background_job_id, times, pauses))
+    logger.warn "#{@background_job_id} wait tracker = #{tracker}, correlation_id = #{tracker.correlation_id}"
+    begin
+      logger.warn "#{@background_job_id}  started = #{tracker.started?}, error = #{tracker.error?}, complete = #{tracker.complete?}, status = #{tracker.status}"
+      logger.warn "Report : #{background_job_report}"
+      sleep(2)
+    end while !tracker.complete?
```
Comment (paulkerschen) [inline lib/torquebox_tester.rb]: @raydavis Although `end while` does parse, my sense is that the Ruby community decided it scans so badly for English speakers that `loop do... break... end` is preferred. http://blade.nagaokaut.ac.jp/cgi-bin/scat.rb/ruby/ruby-core/6745

File: lib/torquebox_tester.rb [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    tracker = background_correlate(background.twiddle_thumbs(@background_job_id, times, pauses))
+    logger.warn "#{@background_job_id} twiddle tracker = #{tracker}, correlation_id = #{tracker.correlation_id}, started = #{tracker.started?}, error = #{tracker.error?}, complete = #{tracker.complete?}, status = #{tracker.status}"
+    logger.warn "Report : #{background_job_report}"
+    "started task #{@background_job_id}"
+  end
+
+  def twiddle_thumbs(test_id, times, pauses)
+    logger.warn "#{test_id} began twiddling"
+    logger.warn "From inside background job Report : #{background_job_report}"
+    for i in 0..(times - 1)
```
Comment (paulkerschen) [inline lib/torquebox_tester.rb]: @raydavis Or (with odd prosody in this instance) `times.times do |i|`

**Comment by raydavis** [comment]
Thank you for the style checks, @paulkerschen -- Destrination... Out!

**Comment by raydavis** [comment]
One happy Travis.

**Comment by paulkerschen** [comment]
So happy he is.

**Pull request merged**